embedded-graphics/src/mock_display.rs

```diff
@@ -166,22 +166,18 @@ where
     }
 }
 
-impl<C> Drawing<C> for MockDisplay<C>
+impl<C> DrawTarget<C> for MockDisplay<C>
 where
     C: PixelColor,
 {
-    fn draw<T>(&mut self, item_pixels: T)
-    where
-        T: IntoIterator<Item = Pixel<C>>,
-    {
-        for Pixel(Point { x, y }, color) in item_pixels {
-            if x < 0 || y < 0 || x >= SIZE as i32 || y >= SIZE as i32 {
-                continue;
-            }
-
-            let i = x + y * SIZE as i32;
-            self.0[i as usize] = Some(color);
+    fn draw_pixel(&mut self, pixel: Pixel<C>) {
+        let Pixel(Point { x, y }, color) = pixel;
+        if x < 0 || y < 0 || x >= SIZE as i32 || y >= SIZE as i32 {
```

Comment:
Just having some fun with Rust syntax:

```suggestion
        if !(0..SIZE).contains(&x) || !(0..SIZE).contains(&y) {
```

This also [optimises out to better assembly](https://godbolt.org/z/5LGkBo). Completely irrelevant for an example so feel free to ignore this comment, I just thought the range check is cool.

Reply:
Woah, nice! I had no idea that was an option. I also have never seen that Compiler Explorer website before-- very cool, I will definitely use that again in the future.

It doesn't hurt to show off some Rust power in examples-- I'll add it in :)